flambe/learn/train.py

```diff
@@ -172,22 +218,66 @@ def _batch_to_device(self, batch: Tuple[torch.Tensor, ...]) -> Tuple[torch.Tenso
 
     def _compute_loss(self, batch: Tuple[torch.Tensor, ...]) -> torch.Tensor:
         """Compute the loss given a single batch
-
+        DEPRECATED, only exists for legacy compatibility with custom trainers
         Parameters
         ----------
         batch: Tuple[torch.Tensor, ...]
             The batch to train on.
 
         """
+        print('Warning: flambe.learn.train.Trainer._compute_loss is deprecated. '
+              'Please use flambe.learn.train.Trainer._compute_batch in the future.')
+        batch = self._batch_to_device(batch)
         pred, target = self.model(*batch)
         loss = self.loss_fn(pred, target)
         return loss
 
+    def _compute_batch(self, batch: Tuple[torch.Tensor, ...], metrics: Optional[list] = ()) \
+            -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
+        """Does a model forward pass over a batch, and returns prediction, target and loss.
+
+        Parameters
+        ----------
+        batch: Tuple[torch.Tensor, ...]
+            The batch to train on.
+
+        """
+        batch = self._batch_to_device(batch)
+        pred, target = self.model(*batch)
+        for metric, state in metrics:
+            metric.aggregate(state, pred, target)
+        loss = self.loss_fn(pred, target)
+        return pred, target, loss
+
+    @staticmethod
+    def _log_metrics(log_prefix: str, metrics_with_states: List[Tuple[Metric, Dict]], global_step: int):
+        """
```

Comment:
wrong docstring format.

Don't jump a line at the start. Should always be:

"""One line description.

More info...

"""

Also keep it under 72 characters